src/useSpring.js

```diff
@@ -1,13 +1,12 @@
 import { useSprings } from './useSprings'
-import { is } from './shared/helpers'
+import { callProp } from './shared/helpers'
 
 /** API
- * const props = useSpring({ ... })
- * const [props, set] = useSpring(() => ({ ... }))
+ * const [props, set, cancel] = useSpring(props, [optionalDeps])
+ * const [props, set, cancel] = useSpring(() => props, [optionalDeps])
  */
 
 export const useSpring = (props, deps) => {
-  const isFn = is.fun(props)
-  const [result, set, stop] = useSprings(1, isFn ? props : [props], deps)
-  return isFn ? [result[0], set, stop] : result
+  const [result, set, stop] = useSprings(1, [callProp(props)], deps)
```

Comment:
This calls the `props` argument on every render. Shouldn't it only be called once?